src/LinAlg/matrix.h

```diff
@@ -1403,9 +1413,12 @@ namespace utl //! General utility classes and functions.
   template<class T> inline
   matrixBase<T>& matrixBase<T>::add(const matrixBase<T>& A, const T& alfa)
   {
+    size_t size = std::min(this->size(),A.size());
+    if (size == 0)
+      return *this;
```

Comment:
Same here, you you replace the below with
```
T* p = this->elem.data();
const T* q = A.elem.data();

Reply:
ok